node/shard/src/lib.rs

```diff
@@ -78,15 +80,11 @@ impl ShardClient {
 
         let chain = Arc::new(chain::BlockChain::new(genesis, storage.clone()));
         let trie_viewer = TrieViewer {};
-        let pool = Arc::new(Pool::new(signer, storage.clone(), trie.clone()));
-        Self { 
-            chain,
-            trie,
-            storage,
-            runtime,
-            trie_viewer,
-            pool
-        }
+        let pool = match signer {
+            Some(signer) => Some(Arc::new(Pool::new(signer, storage.clone(), trie.clone()))),
```

Comment:
This is an implicit logic. We determine whether the pool should be created by checking whether signer is created. Let's make it more explicit by adding a docstring to new that explains that if signer is optional it becomes a non-validating shard client which does not have a pool.